Sources/Tools/plFilePatcher/main.cpp

```diff
@@ -55,7 +55,7 @@ You can contact Cyan Worlds, Inc. by email legal@cyan.com
 #if HS_BUILD_FOR_WIN32
 #   include "hsWindows.h"
 #   include <io.h>
-#   define isatty _isatty
+#   define ttycheck(fileinfo) (GetFileType(fileinfo) == FILE_TYPE_CHAR)
```

Comment:
We prefer to use standards compliant mechanisms. AFAIK, `isatty` is part of the POSIX standard. The Windows SDK just emits an erroneous deprecation warning with `isatty` and wants `_isatty` (which is just the Microsoft way of making life more difficult for cross-platform development). Is there a particular bug that is fixed by using `(GetFileType(fileinfo) == FILE_TYPE_CHAR)` instead of `_isatty`?

Reply:
yes it does if the user attempts compile and run the game using msys2 or cygwin the value of _isatty can be invalid.

Reply:
In that case, you should probably fix the preprocessor checks to detect that situation to use `isatty`.

Reply:
shouldnt HS_BUILD_FOR_WIN32 handle that?

Reply:
`HS_BUILD_FOR_WIN32` will be defined for MSVC, clang-cl, and mingw

In this case, you'll need an extra check for mingw to avoid aliasing isatty

Reply:
This still seems to be using non-standards complaint naming, unfortunately. We prefer the POSIX standard `isassty`.